### DISCUSSION TITLE
Razor components used as Razor pages

### DISCUSSION THREAD

**Discussion by kjeske** [dotnet/aspnetcore#50142] [Q&A] [answered]
I was wondering, would it be possible to use Razor components (.razor) as Razor Pages, which means statically rendered html files served with regular http routed requests? Basically I would like to use @ page directives straight on the .razor files, and get rid of usage of .cshtml files and without any reference to Blazor webassembly or Blazor server. Let's call it: Blazor pages.

Would it require a lot of work to make it happen? Is that in plans?

The reason of my question is the confusion made out of having both kind of files: cshtml and razor, where it seems quite clear that razor files provide much better working experience and are more powerful (eg. slots, RenderFragment).

I know you can statically render the razor component, but you still have to do it from cshtml file and all the routing will be on regular Razor Page.

**Comment by LuohuaRain** [comment, marked as answer]
Blazor web app In ASP.NET Core 8 Preview 7 is what you say.

**Reply by kjeske** [reply]
I don't see it in [the announcement](https://devblogs.microsoft.com/dotnet/asp-net-core-updates-in-dotnet-8-preview-7/#blazor). Could you say something more?

How in ASP.NET Core 8 Preview 7 configure Razor Pages to use .razor files without using .cshtml files?

**Reply by kjeske** [reply]
Ok, I see it now:

```c#
builder.Services.AddRazorComponents();

...

app.MapRazorComponents<App>();
```
